src/ma.jl

```diff
@@ -395,3 +395,34 @@ function vwma(cv::Matrix{T}, n::Int64=10)::Array{Float64} where {T<:Real}
     end
     return out
 end
+
+"""
+```
+vwap(cv::Matrix{T})::Array{T}
+```
+
+volume weighted average price (VWAP)
+"""
+function vwap(cv::Matrix{T})::Array{Float64} where {T<:Real}
```

Comment:
VWAP is generally calculated on intraday prices. Given the absence of timestamps in the function signature, this should probably have some kind of lookback period (e.g. `n::Int=10`).

This appears to be "cumulative" VWMA by default, which is fine as an option but differs from the current convention used by the other functions in this file. I would add another argument after the `n::Int` argument to allow this logic as well, i.e. `cumulative::Bool=false`. This way a lookback period could be used, *or* the current cumulative approach could be used.

As a cool extension, you might also consider adding another function call signature that allows passing a timestamp array (`Vector{TimeType}`) that might group by day and calculate intraday VWAP.

Reply:
Got it I will update the function as per  the requirement